arch/xtensa/src/esp32/esp32_aes.c

```diff
@@ -595,11 +595,6 @@ int aes_cypher(void *out, const void *in, size_t size,
       return -EINVAL;
     }
 
-  if (keysize != 16)
-    {
-      return -EINVAL;
-    }
-
```

Comment:
Did you test using different keysize ? Maybe it should be important to include some safeguard like the maximum keysize allowed